workshops/geometric_pattern/README.md

```diff
@@ -0,0 +1,538 @@
+    for (var x = 0; x <= width; x = x+CIRCLE_DIA) {
+      ellipse(x, 0, CIRCLE_DIA, CIRCLE_DIA);
+    }
+  }
+}
+```
+
+Save and refresh Live Preview to see the change.
+
+### Offsetting the Circles in the X-Direction
+
+It looks pretty cool right now, but our circles aren't overlapping quite right. It looks like every other row should be shifted over by `CIRCLE_RAD`.
+
+We can keep track of alternating rows by using a flag. A flag is just a variable that stores a `true`/`false` value.
+
+Let's add one in our `draw()` function:
+
+```js
+function draw() {
+  var alternatingRow = false;
+  for (var y = 0; y <= height; y = y+CIRCLE_RAD) {
+    for (var x = 0; x <= width; x = x+CIRCLE_DIA) {
+      ellipse(x, 0, CIRCLE_DIA, CIRCLE_DIA);
+    }
+  }
+}
+```
+
+To differentiate between rows, we should add our code inside the outer for-loop, but outside the inner for-loop. This is because the inner for-loop deals with the lineup of circles in the row itself.
+
+We'll flip the value of the flag after each row is created, using the negation operator (`!`):
+
+```js
+function draw() {
+  var alternatingRow = false;
+  for (var y = 0; y <= height; y = y+CIRCLE_RAD) {
+    for (var x = 0; x <= width; x = x+CIRCLE_DIA) {
+      ellipse(x, 0, CIRCLE_DIA, CIRCLE_DIA);
+    }
+    alternatingRow = !alternatingRow;
+  }
+}
+```
+
+This sets the flag `alternatingRow` to its opposite. Its initial value was `false`, and after one row, its value will be `true`. Thus, the second row will be an alternating row. After the second row is created, the flag will be flipped back to false, and so on.
+
+Let's create a conditional that will use the flag's value to determine whether or not to shift the row. We'll add this just before the inner for-loop:
+
+```js
+function draw() {
+  var alternatingRow = false;
+  for (var y = 0; y <= height; y = y+CIRCLE_RAD) {
+    if (alternatingRow) {
+
+    } else {
+
+    }
+    for (var x = 0; x <= width; x = x+CIRCLE_DIA) {
+      ellipse(x, 0, CIRCLE_DIA, CIRCLE_DIA);
+    }
+    alternatingRow = !alternatingRow;
+  }
+}
+```
+
+Now we just need to make this shift. We'll do this by modifying the start `x` value in our inner for-loop. Currently, the start `x` value is always 0, which results in every row starting at the x-coordinate 0.
+
+We can create a variable (`startX`) within `draw()` and use our conditional to set the value of `startX` to 0 or `CIRCLE_RAD`, depending on if it's an alternating row or not. Don't forget to modify the inner for-loop so that the value of `x` starts at `startX`.
+
+```js
+function draw() {
+  var alternatingRow = false;
+  var startX;
+  for (var y = 0; y <= height; y = y+CIRCLE_RAD) {
+    if (alternatingRow) {
+      startX = CIRCLE_RAD;
+    } else {
+      startX = 0;
+    }
+    for (var x = startX; x <= width; x = x+CIRCLE_DIA) {
+      ellipse(x, 0, CIRCLE_DIA, CIRCLE_DIA);
+    }
+    alternatingRow = !alternatingRow;
+  }
+}
+```
+
+Save and check it out!
```

Comment:
I'm finding myself starting to gloss over the writing in the workshop at this point. The code got pretty complex in this section.